File: src/fable-library-py/fable_library/boolean.py
from typing import Any

from .types import FSharpRef


def try_parse(string: str, defValue: FSharpRef[bool]) -> bool:
    string = string.strip().capitalize()
    if string == "True":
        defValue.contents = True
        return True
    elif string == "False":
        defValue.contents = False
        return True

    return False


def parse(value: Any) -> bool:
    def_value: FSharpRef[bool] = FSharpRef(False)
    if try_parse(value, def_value):
        return def_value.contents

    raise ValueError(f"String '{value}' was not recognized as a valid Boolean.")

File: src/fable-compiler-js/fable-library/Boolean.js
import { FSharpRef } from "./Types.js";
export function tryParse(str, defValue) {
    if (str.match(/^\s*true\s*$/i)) {
        defValue.contents = true;
        return true;
    }
    else if (str.match(/^\s*false\s*$/i)) {
        defValue.contents = false;
        return true;
    }
    return false;
}
export function parse(str) {
    const defValue = new FSharpRef(false);
    if (tryParse(str, defValue)) {
        return defValue.contents;
    }
    else {
        throw new Error(`String '${str}' was not recognized as a valid Boolean.`);
    }
}
